### PR TITLE
Add Precheck RPC for various resources

### PR DESCRIPTION
**Pull request by aeweston-msft**
# Changes introduced by this PR
Adding precheck RPC definition for the following resources:
- Identity (security)
- Availability set (compute)
- Gallery image (compute)
- Virtual machine image (compute)
- Load balancer (network)
- Logical network (network)
- MAC pool (network)
- Network interface (network)
- Network security group (network)
- VIP pool (network)
- Virtual network (network)
- Container (storage)

# Test plan
- Have sanity checked the new precheck API for the storage container resource including cloudagent and moc-sdk-for-go changes. Since the Precheck RPC definition is common except for the resource types, I think this is a sufficient minimum of testing.

### CODE REVIEW

**Review by raghavendra-nataraj — APPROVED** [review]

File: rpc/cloudagent/security/identity/moc_cloudagent_identity.proto [hunk 1]
```diff
@@ -40,6 +40,18 @@ message IdentityCertificateResponse {
 	string  Error = 3;
 }
 
+message IdentityPrecheckRequest {
+	repeated Identity Identitys = 1;
```
Comment (aeweston-msft) [inline rpc/cloudagent/security/identity/moc_cloudagent_identity.proto:44]: Should fix to identities

**Review by raghavendra-nataraj — APPROVED** [review]

**Review by ganeshamohana — APPROVED** [review]

**Comment by ganeshamohana** [comment]
LGTM

**Review by apluchik — APPROVED** [review]

**Pull request merged**